### PR TITLE
BRMS-212: Add NRSStatus

### PR DESCRIPTION
**Pull request by chris-ianson**
@manishsw @anujakhare @adamconder 

Added NRSStatus to set status and deathCode if returned.

### CODE REVIEW

File: app/uk/gov/hmrc/brm/models/response/nrs/NRSStatus.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.brm.models.response.nrs
+
+import uk.gov.hmrc.brm.models.response.StatusInterface
+
+case class NRSStatus (
+                       status : Option[Int] = None,
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/models/response/nrs/NRSStatus.scala]: @chris-ianson Should this not be:

`case class NRSStatus(status: Int = 1, deathCode : Int = 0)` for the default?

File: app/uk/gov/hmrc/brm/utils/ReadsUtil.scala [hunk 2]
```diff
@@ -46,16 +45,19 @@ object ReadsUtil {
...
   val groReadRecord : Reads[Record] = (
     JsPath.read[Child](groChildReads) and
       (JsPath \ "status").readNullable[Status]
     )(Record.apply _)
 
   val nrsRecordsRead : Reads[Record] = (
-    (JsPath.read[Child](nrsChildReads)).map(child => Record(child,None))
-    )
-
+    (JsPath.read[Child](nrsChildReads)).map(child => child) and
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/utils/ReadsUtil.scala]: @chris-ianson Do we need to map this now?

**Comment by adamconder** [comment]
@chris-ianson merged. Are we restricting this to a range of numbers?

**Comment by chris-ianson** [comment]
@adamconder Thanks. We're going to pick this up in the other story but yes, we might set it to an Enum

**Comment by adamconder** [comment]
@chris-ianson Which feature?

**Comment by chris-ianson** [comment]
@adamconder In BRMS-213 (handle records found no flags) We could maybe have a 3 amigos when we pull it in and decide if status and deathCode should be restricted by an Enum (we're not actually using them so we need to decide if it's worthwhile work)

**Pull request merged**